wallet-api/src/Wallet/Emulator/Types.hs

```diff
@@ -164,13 +163,37 @@ instance WalletAPI EmulatedWalletApi where
 
 -- Emulator code
 
-type Assertion = EmulatorState -> Maybe AssertionError
+data Assertion
+  = IsValidated Tx
+  | OwnFundsEqual Wallet Value
+
 newtype AssertionError = AssertionError T.Text
     deriving Show
 
-isValidated :: Tx -> Assertion
-isValidated txn emState =
-    if notElem txn (join $ emChain emState) then Just $ AssertionError $ "Txn not validated: " <> T.pack (show txn) else Nothing
+assert :: (MonadEmulator m) => Assertion -> m ()
+assert (IsValidated txn) = isValidated txn
+assert (OwnFundsEqual wallet value) = ownFundsEqual wallet value
+
+ownFundsEqual :: (MonadEmulator m) => Wallet -> Value -> m ()
+ownFundsEqual wallet value = do
+  es <- get
+  ws <- case Map.lookup wallet $ emWalletState es of
+        Nothing -> throwError . AssertionError $ "Wallet not found"
```

Comment:
I'd prefer `throwError $ AssertionError "Wallet not found"` or just `throwError (AssertionError $ "Wallet not found")`.